command/tls_ca_create.go

```diff
@@ -0,0 +1,151 @@
+package command
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/nomad/helper/flags"
+	"github.com/hashicorp/nomad/helper/tlsutil"
+	"github.com/hashicorp/nomad/lib/file"
+	"github.com/posener/complete"
+)
+
+type TLSCACreateCommand struct {
+	Meta
+}
+
+func NewCACreate() *TLSCACreateCommand {
+	return &TLSCACreateCommand{}
+}
+
+func (c *TLSCACreateCommand) Help() string {
+	helpText := `
+Usage: nomad tls ca create [options] filename-prefix
+
+This command has subcommands for interacting with Certificate Authorities.
+
+Here are some simple examples, and more detailed examples are available
+in the subcommands or the documentation.
+
+Create a new Nomad CA
+
+  $ nomad tls ca create
+  ==> saved nomad-agent-ca.pem
+  ==> saved nomad-agent-ca-key.pem
+
+
+CA Create Options:
+  -additional-name-constraint
+    Add name constraints for the CA. Results in rejecting certificates
+    for other DNS than specified. Can be used multiple times. Only used 
+    in combination with -name-constraint.
+
+  -common-name
+    Common Name of CA. Defaults to Nomad Agent CA..
+
+  -days
+    Provide number of days the CA is valid for from now on.
+    Defaults to 5 years or 1825 days..
+
+  -domain
+    Domain of nomad cluster. Only used in combination with -name-constraint. 
+    Defaults to nomad..
+
+  -name-constraint
+    Add name constraints for the CA. Results in rejecting
+    certificates for other DNS than specified. If turned on localhost and 
+    -domain will be added to the allowed DNS. If the Ui is going to be served 
+    over HTTPS its DNS has to be added with -additional-constraint. It is not
+    possible to add that after the fact! Defaults to false..
+
+`
+	return strings.TrimSpace(helpText)
+}
+
+func (c *TLSCACreateCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(c.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-additional-name-constraint": complete.PredictAnything,
+			"-common-name":                complete.PredictAnything,
+			"-days":                       complete.PredictAnything,
+			"-domain":                     complete.PredictAnything,
+			"-name-constraint":            complete.PredictAnything,
+		})
+}
+
+func (c *TLSCACreateCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+func (c *TLSCACreateCommand) Synopsis() string {
+	return "Create Certificate Authority for Nomad"
+}
+
+func (c *TLSCACreateCommand) Name() string { return "tls ca create" }
+
+func (c *TLSCACreateCommand) Run(args []string) int {
+
+	var (
+		days                     int
+		constraint               bool
+		domain                   string
+		commonName               string
+		additionalNameConstraint flags.StringFlag
+	)
+
+	flags := c.Meta.FlagSet(c.Name(), FlagSetClient)
```

Comment:
The flags variable collides with a package import, so we should rename this to something like `flagSet`.